### PR TITLE
GitHub api client issue 24

### PR DESCRIPTION
**Pull request by yitzy32**
## Proposed changes

GithubApi client (class) is a tool that will allow for easy DRY calls to the Github Api. So far this PR covers the `user` and `repositories` methods in the class.

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Refactor (found a better way to write the code and accomplish the same task)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask._

- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by gatorjuice — COMMENTED** [review]
looking good and almost there

File: app/apis/github_api.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+  def user
+    JSON.parse(faraday_client.get(base_uri).body)
+  end
+
+  def repositories(query_params = {})
+    JSON.parse(faraday_client.get("#{base_uri}/repos", query_params).body)
+  end
+
+  def repository(project)
```
[the remark this answers is not in the document]
Comment (yitzy32) [reply app/apis/github_api.rb:21]: Added

[the remark this answers is not in the document]
Comment (gatorjuice) [reply app/apis/github_api.rb:22]: very nice.

**Review by gatorjuice — CHANGES_REQUESTED** [review]
Nice work, this is really nice and simple, but one criteria is still remaining on the story...

* https://github.com/actualize-portfolio/alumni_portfolio_api/blob/main/app/models/github_repo.rb uses this new client as well.

If possible could you address this change, and make sure the github tracker still works as normal?

**Review by gatorjuice — APPROVED** [review]
LGTM!

**Pull request merged**